sys/include/net/ng_netconf.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_netapi Configuration options for network APIs
+ * @ingroup     net
+ * @brief       List of available configuration options for network APIs
+ * @{
+ *
+ * @file
+ * @brief       Definition of global configuration options for netdev and netapi
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NET_CONF_H_
+#define NET_CONF_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Global list of configuration options available throughout the
+ *          network stack, e.g. by netdev and netapi
+ */
+typedef enum {
+    NETCONF_OPT_CHANNEL,            /**< get/set channel as uint16_t in host
+                                         byte order */
+    NETCONF_OPT_IS_CHANNEL_CLR,     /**< check if channel is clear */
+    NETCONF_OPT_ADDRESS,            /**< get/set link layer address in host byte
+                                         order */
+    NETCONF_OPT_ADDRESS_LONG,       /**< long link layer address in host byte
+                                         order (i.e. IEEE802.15.4 EUI-64) */
+    NETCONF_OPT_ADDR_LEN,           /**< get the default address length a
+                                         network device expects as uint16_t in
+                                         host byte order */
+    NETCONF_OPT_NID,                /**< get/set the network ID (e.g. PAN ID in
+                                         IEEE 802.15.4) as uint16_t in host
+                                         byte order */
+    NETCONF_OPT_TX_POWER,           /**< get/set the output power for radio
+                                         devices in dBm as int16_t in host byte
+                                         order */
+    NETCONF_OPT_MAX_PACKET_SIZE,    /**< get the maximum packet size a
```

Comment:
Maybe make this a get/set option, too - for some layers the maximum packet size might be configurable.